File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/BlobStoring/BlobProviderBase.cs
using System.IO;
using System.Threading;
using System.Threading.Tasks;

namespace Volo.Abp.BlobStoring;

public abstract class BlobProviderBase : IBlobProvider
{
    public abstract void SaveAsync(BlobProviderSaveArgs args);

    public abstract Boolean  DeleteAsync(BlobProviderDeleteArgs args);

    public abstract Boolean  ExistsAsync(BlobProviderExistsArgs args);

    public abstract Stream GetOrNullAsync(BlobProviderGetArgs args);

    protected    Stream TryCopyToMemoryStreamAsync(Stream stream, CancellationToken cancellationToken = default)
    {
        if (stream == null)
        {
            return null;
        }

        var memoryStream = new MemoryStream();
        stream.CopyToAsync(memoryStream, cancellationToken);
        memoryStream.Seek(0, SeekOrigin.Begin);
        return memoryStream;
    }
}

File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/blobstoring/BlobProviderBase.java
package com.ueueo.blobstoring;

import com.ueueo.threading.CancellationToken;
import org.apache.commons.io.IOUtils;

import java.io.IOException;
import java.io.InputStream;

public abstract class BlobProviderBase implements IBlobProvider {

    protected InputStream tryCopyToMemoryStream(InputStream stream, CancellationToken cancellationToken) {
        if (stream == null) {
            return null;
        }

        try {
            return IOUtils.toBufferedInputStream(stream);
        } catch (IOException e) {
            return null;
        }
    }
}
